remote-agent/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	tgt "github.com/eclipse-symphony/symphony/api/pkg/apis/v1alpha1/providers/target"
+	"github.com/eclipse-symphony/symphony/api/pkg/apis/v1alpha1/providers/target/script"
+	"github.com/eclipse-symphony/symphony/remote-agent/agent"
+	"github.com/eclipse-symphony/symphony/remote-agent/bindings/http"
+	utils "github.com/eclipse-symphony/symphony/remote-agent/common"
+)
+
+func main() {
+	// Read the configuration file
+	setting, err := ioutil.ReadFile("config.json")
+	if err != nil {
+		fmt.Println("Error reading configuration file:", err)
+		return
+	}
+
+	symphonyEndpoints := utils.SymphonyEndpoint{}
+	err = json.Unmarshal(setting, &symphonyEndpoints)
+	if err != nil {
+		fmt.Println("Error unmarshalling configuration file:", err)
+		return
+	}
+
+	// Compose target providers
+	providers := composeTargetProviders()
+	// Create the HttpBinding instance
+	h := &http.HttpBinding{
+		Agent: agent.Agent{
+			Providers: providers,
+		},
+	}
+
+	// Set up the configuration
+	config := http.HttpBindingConfig{
+		TLS: true,
+		CertProvider: http.CertProviderConfig{
+			Type:   "certs.localfile",
+			Config: map[string]interface{}{"certFile": "path/to/cert.pem", "keyFile": "path/to/key.pem"},
+		},
+	}
+
+	// Set up the request and response URLs
+	requestUrl, err := url.Parse(symphonyEndpoints.RequestEndpoint)
+	if err != nil {
+		fmt.Println("Error parsing request URL:", err)
+		return
+	}
+	responseUrl, err := url.Parse(symphonyEndpoints.ResponseEndpoint)
+	if err != nil {
+		fmt.Println("Error parsing response URL:", err)
+		return
+	}
+	h.RequestUrl = requestUrl
+	h.ResponseUrl = responseUrl
+
+	// Launch the HttpBinding
+	err = h.Launch(config)
+	if err != nil {
+		fmt.Println("Error launching HttpBinding:", err)
+		return
+	}
+
+	// Keep the main function running
+	select {}
+}
+
+func composeTargetProviders() map[string]tgt.ITargetProvider {
+	providers := make(map[string]tgt.ITargetProvider)
+	// Add the target providers to the map
+	// Add the script provider
+	mProvider := &script.ScriptProvider{}
+	providerConfig := script.ScriptProviderConfig{
```

Comment:
Is this a temp version? We need leave this as user input?

Reply:
I think for now, we can define the built-in providers ourselves, not by users